hypothesis-python/src/hypothesis/configuration.py

```diff
@@ -12,22 +12,22 @@
 import sys
 import warnings
 from pathlib import Path
+from typing import Optional
 
 import _hypothesis_globals
 
 from hypothesis.errors import HypothesisSideeffectWarning
 
 __hypothesis_home_directory_default = Path.cwd() / ".hypothesis"
-
 __hypothesis_home_directory = None
 
 
-def set_hypothesis_home_dir(directory):
+def set_hypothesis_home_dir(directory: Optional[os.PathLike]) -> None:
```

Comment:
strictly speaking this isn't correct, eg https://github.com/HypothesisWorks/hypothesis/blob/e182f6917de18163eab91e3f0a2db1c9e0657e65/hypothesis-python/tests/cover/test_settings.py#L279-L280 which passes a str (though mypy won't complain because it's stringified). I would also love to use pathlike but it just won't type correctly.

Reply:
ugh, I always forget that `str` doesn't implement `__fspath__` 😞